packages/grid/x-grid/src/tests/events.XGrid.test.tsx

```diff
@@ -0,0 +1,226 @@
+  // TODO v5: replace with createClientRender
+  const render = createClientRenderStrictMode();
+
+  const baselineProps: { rows: GridRowsProp; columns: GridColumns } = {
+    rows: [
+      {
+        id: 1,
+        first: 'Mike',
+        age: 11,
+      },
+      {
+        id: 2,
+        first: 'Jack',
+        age: 11,
+      },
+      {
+        id: 3,
+        first: 'Mike',
+        age: 20,
+      },
+    ],
+    columns: [
+      { field: 'id' },
+      { field: 'first', editable: true },
+      { field: 'age' },
+      {
+        field: 'firstAge',
+        valueGetter: (params) => `${params.row.first}_${params.row.age}`,
+        valueFormatter: (params) => `${params.value} yrs`,
+      },
+    ],
+  };
+
+  before(function beforeHook() {
+    if (/jsdom/.test(window.navigator.userAgent)) {
+      // Need layouting
+      this.skip();
+    }
+  });
+
+  let apiRef;
+  const TestEvents = (props: Partial<XGridProps>) => {
+    apiRef = useGridApiRef();
+    return (
+      <div style={{ width: 300, height: 300 }}>
+        <XGrid apiRef={apiRef} {...baselineProps} {...props} />
+      </div>
+    );
+  };
+
+  describe('columnHeaderParams', () => {
+    let eventArgs: { params: GridColParams; event: React.MouseEvent } | null = null;
+    let ageColumnElement;
+    beforeEach(() => {
+      const handleClick = (params, event) => {
+        eventArgs = { params, event };
+      };
+      render(<TestEvents onColumnHeaderClick={handleClick} />);
+
+      ageColumnElement = getColumnHeaderCell(3);
+      fireEvent.click(ageColumnElement);
+    });
+
+    it('should include the correct params', () => {
+      expect(eventArgs!.params).to.deep.include({
+        colDef: apiRef!.current.getColumnFromField('age'),
+        element: ageColumnElement,
+        colIndex: 2,
+        field: 'age',
+        api: apiRef.current,
+      });
+    });
+  });
+
+  describe('RowsParams', () => {
+    let eventArgs: { params: GridRowParams; event: React.MouseEvent } | null = null;
+    let row1;
+
+    beforeEach(() => {
+      const handleClick = (params, event) => {
+        eventArgs = { params, event };
+      };
+      render(<TestEvents onRowClick={handleClick} />);
+
+      row1 = getRow(1);
+      fireEvent.click(row1);
+    });
+
+    it('should include the correct params', () => {
+      expect(eventArgs!.params).to.deep.include({
+        id: 2,
+        element: row1,
+        row: baselineProps.rows[1],
+        rowIndex: 1,
+        columns: apiRef!.current.getAllColumns(),
+        api: apiRef.current,
+      });
+    });
+  });
+
+  describe('CellsParams', () => {
+    let eventArgs: { params: GridCellParams; event: React.MouseEvent } | null = null;
+    let cell11;
+
+    beforeEach(() => {
+      const handleClick = (params, event) => {
+        eventArgs = { params, event };
+      };
+      render(<TestEvents onCellClick={handleClick} />);
+
+      cell11 = getCell(1, 1);
+      fireEvent.click(cell11);
```

Comment:
This before each actions seem only relevant for the first it. could we move it there? The more isolated each test, the better.